### PR TITLE
Add assets with backfill policies to toys

### PR DESCRIPTION
**Pull request by clairelin135**
As the title.

Adds to toys to make local testing / canary testing easier.

### CODE REVIEW

**Comment by clairelin135** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #18980** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18980?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18980?utm_source=stack-comment).

File: python_modules/dagster-test/dagster_test/toys/partitioned_assets/with_backfill_policies.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from dagster import BackfillPolicy, DailyPartitionsDefinition, MaterializeResult, asset
+
+
+@asset(
+    partitions_def=DailyPartitionsDefinition(start_date="2023-01-01"),
+    backfill_policy=BackfillPolicy.single_run(),
+    group_name="with_backfill_policy",
+)
+def successful_daily_asset() -> MaterializeResult:
```
Comment (sryza) [inline python_modules/dagster-test/dagster_test/toys/partitioned_assets/with_backfill_policies.py:9]: maybe call this `single_run_backfill_policy` or `single_run_backfill_policy`? for someone browsing assets in go/toys, that will make it clear what this one is there for.

Comment (clairelin135) [reply python_modules/dagster-test/dagster_test/toys/partitioned_assets/with_backfill_policies.py]: updated!

**Review by sryza — APPROVED** [review]
nice

**Pull request merged**